runtime/onert/core/include/backend/IConfig.h

```diff
@@ -69,6 +69,8 @@ struct IConfig
   virtual bool supportPermutation() = 0;
   virtual bool supportDynamicTensor() = 0;
   virtual bool supportFP16() = 0;
+  // TODO Find a way to check whether the backend supports training at compile time
+  virtual bool supportTraining() { return false; }
```

Comment:
Now, I think the only way to enter the training mode is to call the `prepare_train()` function like [this](https://github.com/Samsung/ONE/pull/10825/files#diff-c62b86f62bfb800409860fb5a02724a75abbd4d95c6ce2f2833bb3353caf5288).
I'm not sure if it supports all cases for training mode. And I don't know which is better, `Config` or `API`.

Reply:
This is introduced to check whether a backend supports training. Unfortunately, we cannot check it just by checking whether a session has the intention to train.
Whether or not a backend supports training is determined in the code, but since this method is checked at runtime, I want to find a better way.

Reply:
I introduced the `prepare_train()` function. It can help to determine whether this session is trainable or not. How about it?